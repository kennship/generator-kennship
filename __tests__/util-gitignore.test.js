const { mergeGitignore } = require('../generators/util/gitignore');

describe('util/gitignore', () => {
  it('should merge two configs', () => {
    const original = `
yourfile.txt
myfile.txt

# npm
node_modules
*.log
npm-debug.log*

# editors
*.sw*`.split('\n');

    const newRules = `
myfile.txt
ourfile.txt

# terraform
.terraform
*.tfstate
*.tfvars

# editors
.*~

# assorted
node_modules
.npm`.split('\n');

    const result = mergeGitignore(original, newRules).join('\n');
    expect(result).toMatchSnapshot();
  });
});
