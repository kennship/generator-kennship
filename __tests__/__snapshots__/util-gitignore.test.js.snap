// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`util/gitignore should merge two configs 1`] = `
"myfile.txt
ourfile.txt
yourfile.txt

# npm
*.log
.npm
node_modules
npm-debug.log*

# editors
*.sw*
.*~

# terraform
*.tfstate
*.tfvars
.terraform
"
`;
